### DISCUSSION TITLE
Dynamically creating public properties in component

### DISCUSSION THREAD

**Discussion by biacho** [livewire/livewire#6684] [1. Help]
Hi guys, 

I have a question, is there a good way to create dynamically public properties in component class?
I want to create generic form component, based on db table editable by user in admin panel. So user will create forms fields and also dependencies between those fields (for ex. based on chosen option in select tag other field will be hidden or showing), i could used in this case JS, but I'm curious if livewire is able do this.

Thanks,

**Comment by joshhanley** [comment]
@biacho PHP has now deprecated dynamic properties in classes, so the best thing to do would be to use an array or collection to store your data, with the keys being your dynamic property name.

Hope this helps!
